<a href="https://colab.research.google.com/github/Ivysot/Alura-Curso1/blob/main/Hora%2Bda%2Bpratica.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sobre o notebook

A "hora da prática" é uma atividade do curso de Python para Data Science da Alura que contém diversos desafios envolvendo os aprendizados de cada aula.

O notebook "hora da pratica" é um notebook construído com espaço para as células dos desafios propostos no curso, com ele você tem espaço para construir e executar suas soluções

## Aula 1 - Introdução ao Python

### Questão 1

Imprima a frase `Escola de Dados da Alura!`

In [ ]:
print('Escola de Dados da Alura!')

Escola de Dados da Alura!


### Questão 2
Imprima na tela seu nome e seu sobrenome seguindo a estrutura abaixo:
```
Nome: [seu nome]
Sobrenome: [seu sobrenome]
```


In [ ]:
print('Nome: Vitor')
print('Sobrenome: Soyama')

Nome: Vitor
Sobrenome: Soyama


### Questão 3
Imprima o seu primeiro nome letra a letra. Por exemplo, nome é Mirla, então eu obtenho a seguinte saída:
```
M
I
R
L
A
```


In [ ]:
print('V')
print('I')
print('T')
print('O')
print('R')

V
I
T
O
R


### Questão 4
Imprima o dia do seu nascimento em formato `dia mês ano`. Os valores de dia e ano não podem estar entre aspas. O formato deve estar como no exemplo, supondo uma data de aniversário dia 28 de fevereiro de 2003:
```
28 fevereiro 2003
```

In [ ]:
data = 9
ano = 1998


print( str(data) + ' de Junho de ' + str(ano))

9 de Junho de 1998


### Questão 5
Imprima em um único print o atual ano que você está fazendo esse curso, o valor do ano deve ser um dado numérico. A saída do print deve ser a seguinte:
```
Ano atual: [ano]
```


In [ ]:
anoAtual = 2023
print(f'Ano atual: {anoAtual}')

Ano atual: 2023


## Aula 2 - Manipulando dados no Python

### Coleta e amostragem de dados

#### Questão 1
Crie um programa que solicite à pessoa usuária digitar seu nome, e imprima “Olá, [nome]!”.

In [ ]:
nomeInput = input("Digite seu nome:")
print("Olá, "+ nomeInput)

Digite seu nome:Vitor
Olá, Vitor


#### Questão 2
Crie um programa que solicite à pessoa usuária digitar seu nome e idade, e imprima “Olá, [nome], você tem [idade] anos.”.

In [ ]:
nomeInput = input("Digite seu nome:")
idadeInput = input("Digite sua idade:")

print(f"Olá, {nomeInput}, você tem {idadeInput} anos ")

Digite seu nome:Vitor
Digite sua idade:25
Olá, Vitor, você tem 25 anos 


#### Questão 3
Crie um programa que solicite à pessoa usuária digitar seu nome, idade e altura em metros, e imprima “Olá, [nome], você tem [idade] anos e mede [altura] metros!”.

In [ ]:
nomeInput = input("Digite seu nome:")
idadeInput = input("Digite sua idade:")
alturaInput = input("Digite sua altura:")

print(f"Olá, {nomeInput}, você tem {idadeInput} anos e mede {alturaInput} metros!")

Digite seu nome:Vitor
Digite sua idade:25
Digite sua altura:1,71
Olá, Vitor, você tem 25 anos e mede 1,71 metros!


### Calculadora com operadores

#### Questão 1

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a soma dos dois valores.

In [ ]:
n1 = input("Digite o primeiro numero: ")
n2 = input("Digite o segundo numero: ")

soma = int(n1)+int(n2)

print(f'soma: {soma}' )

Digite o primeiro numero: 1
Digite o segundo numero: 2
soma: 3


#### Questão 2
Crie um programa que solicite três valores numéricos à pessoa usuária e imprima a soma dos três valores.

In [ ]:
n1 = float(input("Digite o primeiro numero: "))
n2 = float(input("Digite o segundo numero: "))
n3 = float(input("Digite o terceiro numero: "))

soma = n1+n2+n3

print(f'soma: {soma}' )

Digite o primeiro numero: 1.2
Digite o segundo numero: 1.4
Digite o terceiro numero: 1.5
soma: 4.1


#### Questão 3
Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a subtração do primeiro pelo o segundo valor.

In [ ]:
n1 = float(input("Digite o primeiro numero: "))
n2 = float(input("Digite o segundo numero: "))

subtracao = n1-n2

print(f'subtração: {subtracao}' )

Digite o primeiro numero: 1.20
Digite o segundo numero: 1.0
subtração: 0.19999999999999996


#### Questão 4

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a multiplicação dos dois valores.


In [2]:
n1 = float(input("Digite o primeiro numero: "))
n2 = float(input("Digite o segundo numero: "))

subtracao = n1*n2

print(f'multiplicação: {subtracao}' )

Digite o primeiro numero: 244
Digite o segundo numero: 400
multiplicação: 97600.0


#### Questão 5

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e realize a divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.


In [3]:
n1 = float(input("Digite o numerador: "))
n2 = float(input("Digite o denominador diferente de zero: "))

subtracao = n1/n2

print(f'divisão: {subtracao}' )

Digite o numerador: 8
Digite o denominador diferente de zero: 4
divisão: 2.0


#### Questão 6

Crie um programa que solicite dois valores numéricos, um operador e uma potência, e realize a exponenciação entre esses dois valores.


In [5]:
n1 = float(input("Digite o operador: "))
n2 = float(input("Digite a potência: "))

subtracao = n1**n2

print(f'exponenciação: {subtracao}' )

Digite o operador: 4
Digite a potência: 10
exponenciação: 1048576.0


#### Questão 7
Crie um programa que solicite dois valores numéricos, um numerador e um denominador e realize a divisão inteira entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [7]:
n1 = int(input("Digite o numerador: "))
n2 = int(input("Digite o denominador diferente de zero: "))

subtracao = n1//n2

print(f'divisão: {subtracao}' )

Digite o numerador: 5
Digite o denominador diferente de zero: 2
divisão: 2


#### Questão 8

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e retorne o resto da divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [8]:
n1 = int(input("Digite o numerador: "))
n2 = int(input("Digite o denominador diferente de zero: "))

subtracao = n1%n2

print(f'divisão: {subtracao}' )

Digite o numerador: 5
Digite o denominador diferente de zero: 2
divisão: 1


#### Questão 9

Crie um código que solicita 3 notas de um estudante e imprima a média das notas.

In [10]:
n1 = float(input("Digite a primeira nota: "))
n2 = float(input("Digite a segunda nota: "))
n3 = float(input("Digite a terceira nota: "))

media = (n1+n2+n3)/3

print(f'média: {media}' )

Digite a primeira nota: 1
Digite a segunda nota: 2
Digite a terceira nota: 3
média: 2.0


#### Questão 10

Crie um código que calcule e imprima a média ponderada dos números 5, 12, 20 e 15 com pesos respectivamente iguais a 1, 2, 3 e 4.

In [11]:
n1 = 5
n2 = 12
n3 = 20
n4 = 15

media = (n1*1+n2*2+n3*3+n4*4)/10

print(f'média ponderada: {media}' )

média ponderada: 14.9


### Editando textos

#### Questão 1

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase na tela.

In [13]:
frase = 'Olá mundo!'
print(frase)

Olá mundo!


#### Questão 2
Crie um código que solicite uma frase e depois imprima a frase na tela.

In [14]:
frase = input("Digite uma frase:")
print(frase)

Digite uma frase:Olá
Olá


#### Questão 3

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras maiúsculas.

In [15]:
frase = input("Digite uma frase:")
frase = frase.upper()
print(frase)

Digite uma frase:Olá
OLÁ


#### Questão 4

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras minúsculas.

In [16]:
frase = input("Digite uma frase:")
frase = frase.lower()
print(frase)

Digite uma frase:Olá
olá


#### Questão 5

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase sem espaços em branco no início e no fim.

In [17]:
frase = '    Olá mundo!   '
frase = frase.strip(" ")

print(frase)


Olá mundo!


#### Questão 6

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim.

In [18]:
frase = input('Digite uma frase:')
frase = frase.lstrip(" ")


print(frase)


Digite uma frase:     Olá Mundo!      
Olá Mundo!


#### Questão 7


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim e em letras minúsculas.

In [19]:
frase = input('Digite uma frase:')
frase = frase.strip(" ")
frase = frase.lower()

print(frase)

Digite uma frase:          OLÀ MUNDO!        
olà mundo!


#### Questão 8

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`e`” trocadas pela letra “`f`”.

In [20]:
frase = input('Digite uma frase:')
frase = frase.replace("e", "f")


print(frase)

Digite uma frase:estou aqui
fstou aqui


#### Questão 9


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`a`” trocadas pela caractere  “`@`”.

In [22]:
frase = input('Digite uma frase:')
frase = frase.replace("a", "@")


print(frase)

Digite uma frase:aqui estou
@qui estou


#### Questão 10


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as consoantes “`s`” trocadas pelo caractere  “`$`”.

In [23]:
frase = input('Digite uma frase:')
frase = frase.replace("s", "$")


print(frase)

Digite uma frase:estou aqui
e$tou aqui


## Aula 3 - Estruturas condicionais

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça à pessoa usuária para fornecer dois números e exibir o número maior.

In [28]:
n1 = float(input('Digite um número:'))
n2 = float(input('Digite outro número:'))

if(n1>n2):
  print(f'O número: {n1} é maior')
elif(n2>n1):
  print(f'O número: {n2} é maior')
else:
  print('Os dois números são iguais')


Digite um número:2
Digite outro número:1
O número: 2.0 é maior


#### Questão 2
Escreva um programa que solicite o percentual de crescimento de produção de uma empresa e informe se houve um crescimento (porcentagem positiva) ou decrescimento (porcentagem negativa).

In [26]:
# Coleta do percentual
percentual = float(input('Digite o percentual de crescimento: '))


if percentual > 0:
    print(f'Houve um crescimento de {percentual}%')
elif percentual < 0:
    print(f'Houve um decrescimento de {percentual}%')
else:
    print('Não houve crescimento ou decrescimento.')

Digite o percentual de crescimento: -1
Houve um decrescimento de -1.0%


#### Questão 3
Escreva um programa que determine se uma letra fornecida pela pessoa usuária é uma vogal ou consoante.

In [31]:
letra = input('Digite uma letra:')

vogais = 'aeiou'

if (letra in vogais):
  print("É uma vogal")
else:
  print('É uma consoante')


Digite uma letra:a
É uma vogal


#### Questão 4
Escreva um programa que leia valores médios de preços de um modelo de carro por 3 anos consecutivos e exiba o valor mais alto e mais baixo entre esses três anos.

In [32]:

preco_ano1 = float(input('Informe o preço médio do carro no primeiro ano: '))
preco_ano2 = float(input('Informe o preço médio do carro no segundo ano: '))
preco_ano3 = float(input('Informe o preço médio do carro no terceiro ano: '))


if preco_ano2 > preco_ano1 and preco_ano2 > preco_ano3:
  maior = preco_ano2
elif preco_ano3 > preco_ano2 and preco_ano3 > preco_ano1:
  maior = preco_ano3
else:
  maior = preco_ano1


if preco_ano2 < preco_ano1 and preco_ano2 < preco_ano3:
  menor = preco_ano2
elif preco_ano3 < preco_ano1 and preco_ano3 < preco_ano2:
  menor = preco_ano3
else:
  menor = preco_ano1


print(f'O preço mais alto foi de R$ {maior}.')
print(f'O preço mais baixo foi de R$ {menor}.')

Informe o preço médio do carro no primeiro ano: 1000
Informe o preço médio do carro no segundo ano: 1500
Informe o preço médio do carro no terceiro ano: 2000
O preço mais alto foi de R$ 2000.0.
O preço mais baixo foi de R$ 1000.0.


#### Questão 5
Escreva um programa que pergunte sobre o preço de três produtos e indique qual é o produto mais barato para comprar.

In [33]:
p1 = float(input('Informe o preço do produto 1: '))
p2 = float(input('Informe o preço do produto 2: '))
p3 = float(input('Informe o preço do produto 3: '))


if p1 < p2 and p1 < p3:
  menor = 1
elif p2 < p1 and p2 < p3:
  menor = 2
else:
  menor = 3


print(f'O produto mais barato é o produto {menor}.')

Informe o preço do produto 1: 10
Informe o preço do produto 2: 20
Informe o preço do produto 3: 15
O produto mais barato é o produto 1.


#### Questão 6
Escreva um programa que leia três números e os exiba em ordem decrescente.

In [34]:
n1 = float(input("Digite o primeiro número:"))
n2 = float(input("Digite o segundo número:"))
n3 = float(input("Digite o terceiro número:"))

if n1 <= n2 and n1 <= n3:
  menor = n1
  if n2 >= n3:
    maior = n2
    meio = n3
  else:
    maior = n3
    meio = n2
elif n1 <= n1 and n1 >= n3:
  meio = n1
  if n2 >= n3:
    maior = n2
    menor = n3
  else:
    maior = n3
    menor = n2
else:
  maior = n1
  if n2 >= n3:
    meio = n2
    menor = n3
  else:
    meio = n3
    menor = n2

print(f'{maior}; {meio}; {menor}')





Digite o primeiro número:1
Digite o segundo número:2
Digite o terceiro número:3
3.0; 2.0; 1.0


#### Questão 7
Escreva um programa que pergunte em qual turno a pessoa usuária estuda ("manhã", "tarde" ou "noite") e exiba a mensagem "Bom Dia!", "Boa Tarde!", "Boa Noite!", ou "Valor Inválido!", conforme o caso.

In [36]:
turno = input("Por favor informe o seu turno(manhã, tarde ou noite): ")

turno = turno.lower()

if turno == "manhã":
  print("Bom dia!")
elif turno == "tarde":
  print("Boa tarde!")
elif turno == "noite":
  print("Boa noite!")
else:
  print("Valor Inválido")


Por favor informe o seu turno(manhã, tarde ou noite)Manhã
Bom dia!


#### Questão 8
Escreva um programa que peça um número inteiro à pessoa usuária e determine se ele é par ou ímpar. Dica: Você pode utilizar o operador módulo `%`.

In [2]:
numero = int(input('Digite um número:'))

resto = numero%2

if resto != 0:
  print("O número é impar")
else:
  print("O número é par")

Digite um número:51
O número é impar


#### Questão 9
Escreva um programa que peça um número à pessoa usuária e informe se ele é inteiro ou decimal.

In [5]:
numero = float(input('Digite um número:'))

resto = numero%1

if resto != 0:
  print("O número é decimal")
else:
  print("O número é inteiro")

Digite um número:1.0
O número é inteiro


### Momento dos projetos

#### Questão 10
Um programa deve ser escrito para ler dois números e, em seguida, perguntar à pessoa usuária qual operação ele deseja realizar. O resultado da operação deve incluir informações sobre o número - se é par ou ímpar, positivo ou negativo e inteiro ou decimal.

In [10]:
n1 = float(input('Digite o primeiro número: '))

n2 = float(input('Digite o segundo número: '))

acao = input('Qual ação gostaria de realizar (+, -, *, /): ')


if acao == '+':
  resultado = n1+n2
elif acao == '-':
  resultado = n1-n2
elif acao == '*':
  resultado = n1*n2
elif acao == '/':
  resultado = n1/n2
else:
  print('Ação inválida')
  resultado = ''

if resultado != '':
  restoInt = resultado%1
  restoPar = resultado%2

  if restoInt != 0:
    inteiro = "Decimal"
  else:
    inteiro = "Inteiro"

  if restoPar != 0:
    par = "Ímpar"
  else:
    par = "Par"

  if resultado >= 0:
    pos = "Positivo"
  else:
    pos = "Negativo"
  print(f'O resultado é igual à {resultado}, e é {inteiro}, {par} e {pos}')


Digite o primeiro número: 8
Digite o segundo número: 8
Qual ação gostaria de realizar (+, -, *, /): 8
Ação inválida


#### Questão 11
Escreva um programa que peça à pessoa usuária três números que representam os lados de um triângulo. O programa deve informar se os valores podem ser utilizados para formar um triângulo e, caso afirmativo, se ele é equilátero, isósceles ou escaleno. Tenha em mente algumas dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;


In [7]:
lado1 = float(input("Digite o lado 1: "))
lado2 = float(input("Digite o lado 2: "))
lado3 = float(input("Digite o lado 3: "))

if lado1+lado2 < lado3 or lado1+lado3 < lado2 or lado2+lado3 < lado1:
  print("Triângulo inválido!")
else:
  if lado1 == lado2 == lado3:
    print("É um triângulo Equilátero")
  elif lado1 != lado2 != lado3:
    print("É um triângulo Escaleno")
  else:
    print("É um triângulo Isósceles")


Digite o lado 1: 6
Digite o lado 2: 6
Digite o lado 3: 5
É um triângulo Isósceles


#### Questão 12
Um estabelecimento está vendendo combustíveis com descontos variados. Para o etanol, se a quantidade comprada for até 15 litros, o desconto será de 2% por litro. Caso contrário, será de 4% por litro. Para o diesel, se a quantidade comprada for até 15 litros, o desconto será de 3% por litro. Caso contrário, será de 5% por litro. O preço do litro de diesel é R\$ 2,00 e o preço do litro de etanol é R\$ 1,70. Escreva um programa que leia a quantidade de litros vendidos e o tipo de combustível (E para etanol e D para diesel) e calcule o valor a ser pago pelo cliente. Tenha em mente algumas dicas:

- O do valor do desconto será a multiplicação entre preço do litro, quantidade de litros e o valor do desconto.
- O valor a ser pago por um cliente será o resultado da multiplicação do preço do litro pela quantidade de litros menos o valor de desconto resultante do cálculo.


In [13]:
combustivel = input("Digite o tipo de combustível(Etanol(E) ou Diesel(D)):")

if combustivel.lower() != "d" and combustivel.lower() != "e":
  print("Combustível Inválido")
else:
  litros = float(input("Digite a quantidade de litros que deseja abastecer:"))
  if litros > 0:
    if combustivel == "E":
      precoL = 1.70
      if litros < 15:
        desconto = 0.02
      else:
        desconto = 0.04
    if combustivel == "D":
      precoL = 2.00
      if litros < 15:
        desconto = 0.03
      else:
        desconto = 0.05
  else:
    print("Quantidade de litros inválida")

  precoF = precoL*litros*(1-desconto)

  print(f"O total a pagar será: {precoF} com o desconto de {desconto*100}% e preço o litro de: {precoL}")



Digite o tipo de combustível(Etanol(E) ou Diesel(D)):D
Digite a quantidade de litros que deseja abastecer:15
O total a pagar será: 28.5 com o desconto de 5.0% e preço o litro de: 2.0


#### Questão 13
Em uma empresa de venda de imóveis você precisa criar um código que analise os dados de vendas anuais para ajudar a diretoria na tomada de decisão. O código precisa coletar os dados de quantidade de venda durante os anos de 2022 e 2023 e fazer um cálculo de variação percentual. A partir do valor da variação, deve ser enviada às seguintes sugestões:
- Para variação acima de 20%: bonificação para o time de vendas.
- Para variação entre 2% e 20%: pequena bonificação para time de vendas.
- Para variação entre 2% e -10%: planejamento de políticas de incentivo às vendas.
- Para bonificações abaixo de -10%: corte de gastos.


In [ ]:

venda2022 = float(input('Informe a quantidade de vendas em 2022: '))
venda2023 = float(input('Informe a quantidade de vendas em 2023: '))


varPercentual = 100 * (venda2023 - venda2022) / (venda2022)


if varPercentual > 20:
    print('Bonificação para o time de vendas.')
elif 2 <= varPercentual <= 20:
    print('Pequena bonificação para time de vendas.')
elif -10 <= varPercentual < 2:
    print('Planejamento de políticas de incentivo às vendas.')
else:
    print('Corte de gastos.')

## Aula 4 - Estruturas de repetição

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça dois números inteiros e imprima todos os números inteiros entre eles.

In [19]:
n1 = int(input("Digite o primeiro número:"))
n2 = int(input("Digite o segundo número"))

if n1 > n2:
  maior = n1
  menor = n2
else:
  maior = n2
  menor = n1

for i in range(menor+1, maior):
  print(i)

Digite o primeiro número:5
Digite o segundo número1
2
3
4


#### Questão 2
Escreva um programa para calcular quantos dias levará para a colônia de uma bactéria A ultrapassar ou igualar a colônia de uma bactéria B, com base nas taxas de crescimento de 3% e 1,5% respectivamente. Considere que a colônia A inicia com 4 elementos e a colônia B com 10 elementos.

In [24]:
popA = 4
popB = 10

taxaA = 3
taxaB = 1.5

dias = 0

while popA <= popB:
  popA *= (1+(taxaA/100))
  popB *= (1+(taxaB/100))
  dias += 1

print(f"Demorará {dias} dias")

print(popA)
print(popB)

Demorará 63 dias
25.75165514029839
25.548220848110983


#### Questão 3
Para tratar uma quantidade de 15 dados de avaliações de pessoas usuárias de um serviço da empresa, precisamos verificar se as notas são válidas. Então, escreva um programa que vai receber a nota de 0 a 5 de todos os dados e verificar se é um valor válido. Caso seja inserido uma nota acima de 5 ou abaixo de 0, repita até que a pessoa usuária insira um valor válido.

#### Questão 4
Desenvolva um programa que leia um conjunto indeterminado de temperaturas em Celsius e faça uma análise. Portanto, escreva um programa que leia temperaturas e informe a média delas. A leitura deve ser encerrada ao ser enviado o valor -273°C.

#### Questão 5
Escreva um programa que calcule o fatorial de um número inteiro fornecido pela pessoa usuária. O fatorial de um número inteiro é a multiplicação desse número por todos os seus antecessores até o número 1. Por exemplo, o fatorial de 5 é 5 x 4 x 3 x 2 x 1 = 120.

### Momento dos projetos

#### Questão 6
Escreva um programa que gere a tabuada de um número inteiro de 1 a 10, de acordo com a escolha da pessoa usuária. Como exemplo, para o número 2, a tabuada deve ser mostrada no seguinte formato:
```
Tabuada do 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20
```

#### Questão 7
Os números primos possuem várias aplicações dentro da Ciência de Dados, por exemplo, na criptografia e segurança. Um número primo é aquele que é divisível apenas por um e por ele mesmo. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.

#### Questão 8
Vamos entender a distribuição de idades de pensionistas de uma empresa de previdência. Escreva um programa que leia as idades de uma quantidade não informada de clientes e mostre a distribuição em intervalos de [0-25], [26-50], [51-75] e [76-100]. Encerre a entrada de dados com um número negativo.

#### Questão 9
Em uma eleição para gerência em uma empresa com 20 funcionários, existem quatro candidatos. Escreva um programa que calcule o vencedor da eleição. A votação ocorreu da seguinte maneira:
- Cada funcionário votou em um dos quatro candidatos (representados pelos números 1, 2, 3 e 4).
- Também foram contabilizados os votos nulos (representado pelo número 5) e os votos em branco (representado pelo número 6).

Ao final da votação, o programa deve exibir o total de votos para cada candidato, o número de votos nulos e o número de votos em branco. Além disso, deve calcular e exibir a porcentagem de votos nulos em relação ao total de votos e a porcentagem de votos em branco em relação ao total de votos.

## Aula 5 - Estruturas de dados

### Aquecendo na programação

#### Questão 1
Faça um programa que tenha a seguinte lista contendo os valores de gastos de uma empresa de papel `[2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]`. Com esses valores, faça um programa que calcule a média de gastos. Dica, use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).

#### Questão 2
 Com os mesmos dados da questão anterior, defina quantas compras foram acima de 3000 reais e calcule a porcentagem quanto ao total de compras.

#### Questão 3
Faça um código que colete em uma lista 5 números inteiros quaisquer e imprima a lista. Exemplo: `[1,4,7,2,4]`.

#### Questão 4
Colete novamente 5 inteiros e imprima a lista em ordem inversa à enviada.

#### Questão 5
Faça um programa que, ao inserir um número qualquer, criará uma lista contendo todos os números primos entre 1 e o número digitado.

#### Questão 6
Escreva um programa que peça uma data informando o dia, mês e ano e determine se ela é uma data válida para uma análise.

### Momento dos projetos

#### Questão 7
Para um estudo envolvendo o nível de multiplicação de bactérias em uma colônia, foi coletado o número de bactérias multiplicadas por dia, com base em um processo de duplicação diária e pode ser observado a seguir: `[1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]`. Tendo esses valores, faça um código que gere uma lista contendo o percentual de crescimento de bactérias por dia, comparando o número de bactérias em cada dia com o número de bactérias do dia anterior. Dica: para calcular o percentual de crescimento usamos a seguinte equação: `100 * (amostra_atual - amostra_passada) / (amostra_passada)`.

#### Questão 8
Para uma seleção de produtos alimentícios, precisamos separar o conjunto de IDs dados por números inteiros, sabendo que os produtos com ID par são doces e os com ID ímpar são amargos. Monte um código que colete 10 IDs. Depois, calcule e mostre a quantidade de produtos doces e amargos.

#### Questão 9
Desenvolva um programa que informa a nota de um aluno de acordo com suas respostas. Ele deve pedir a resposta de um aluno para cada questão e é preciso verificar se a resposta foi igual ao gabarito. Cada questão vale um ponto e existem alternativas:  A, B, C ou D.
```
Gabarito da Prova:
01 - D
02 - A
03 - C
04 - B
05 - A
06 - D
07 - C
08 - C
09 - A
10 - B
```

#### Questão 10
Um instituto de meteorologia deseja fazer um estudo de temperatura média de cada mês do ano. Para isso, você precisa fazer um código que colete e armazene essas temperaturas médias em uma lista. Depois, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês elas ocorreram, mostrando os meses por extenso: Janeiro, Fevereiro, etc.

#### Questão 11
Uma empresa de e-commerce está interessada em analisar as vendas dos seus produtos. Os dados das vendas foram armazenados em um dicionário:
```
{'Produto A': 300, 'Produto B': 80, 'Produto C': 60,
 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}
```
Escreva um código que calcule o total de vendas e o produto mais vendido.

#### Questão 12
Uma pesquisa de mercado foi feita para decidir qual design de uma marca infantil mais agrada crianças. A pesquisa foi feita e o votos computados podem ser observados abaixo:
```
'''
Tabela de votos da marca
Design 1 - 1334 votos
Design 2 - 982 votos
Design 3 - 1751 votos
Design 4 - 210 votos
Design 5 - 1811 votos
'''
```
Adapte os dados fornecidos a você para uma estrutura de dicionário e a partir dele, informe o design vencedor e a porcentagem de votos recebidos.


#### Questão 13
Os funcionários de um setor da empresa que você trabalha vão receber um abono correspondente a 10% do seu salário devido ao ótimo desempenho do time. O setor financeiro solicitou sua ajuda para a verificação das consequências financeiras que esse abono irá gerar nos recursos. Assim, foi encaminhada para você uma lista com os salários que receberão o abono: `[1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]`. O abono de cada funcionário não pode ser inferior a 200. Em código, transforme cada um dos salários em chaves de um dicionário e o abono de cada salário no elemento. Depois, informe o total de gastos com o abono, quantos funcionários receberam o abono mínimo e qual o maior valor de abono fornecido.


#### Questão 14
Uma equipe de cientistas de dados está estudando a diversidade biológica em uma floresta. A equipe fez a coleta de informações sobre o número de espécies de plantas e animais em cada área da floresta e armazenaram essas informações em um dicionário. Nele, a chave descreve a área dos dados e os valores nas listas correspondem às espécies de plantas e animais nas áreas, respectivamente.
```
{'Área Norte': [2819, 7236],
 'Área Leste': [1440, 9492],
 'Área Sul': [5969, 7496],
 'Área Oeste': [14446, 49688],
 'Área Centro': [22558, 45148]}
```
Escreva um código para calcular a média de espécies por área e identificar a área com a maior diversidade biológica. Dica: use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).


#### Questão 15
O setor de RH da sua empresa te pediu uma ajuda para analisar as idades dos funcionários de 4 setores da empresa. Para isso, ele te forneceu os seguintes dados:
```
{'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
```
Sabendo que cada setor tem 10 funcionários, construa um código que calcule a média de idade de cada setor, a idade média geral entre todos os setores e quantas pessoas estão acima da idade média geral.
